blockly-ws-search/src/WorkspaceSearch.js

```diff
@@ -0,0 +1,655 @@
+    /**
+     * The workspace the search bar sits in.
+     * @type {!Blockly.WorkspaceSvg}
+     * @private
+     */
+    this.workspace_ = workspace;
+
+    /**
+     * HTML container for the search bar.
+     * @type {?HTMLElement}
+     * @private
+     */
+    this.htmlDiv_ = null;
+
+    /**
+     * The div that holds the search bar actions.
+     * @type {?HTMLElement}
+     * @protected
+     */
+    this.actionDiv_ = null;
+
+    /**
+     * The text input for the search bar.
+     * @type {?HTMLInputElement}
+     * @private
+     */
+    this.inputElement_ = null;
+
+    /**
+     * The placeholder text for the search bar input.
+     * @type {string}
+     * @private
+     */
+    this.textInputPlaceholder_ = 'Search';
+
+    /**
+     * A list of blocks that came up in the search.
+     * @type {!Array.<Blockly.BlockSvg>}
+     * @protected
+     */
+    this.blocks_ = [];
+
+    /**
+     * Index of the currently "selected" block in the blocks array.
+     * @type {number}
+     * @protected
+     */
+    this.currentBlockIndex_ = -1;
+
+    /**
+     * The search text.
+     * @type {string}
+     * @protected
+     */
+    this.searchText_ = '';
+
+    /**
+     * Whether to search as input changes as opposed to on enter.
+     * @type {boolean}
+     */
+    this.searchOnInput = true;
+
+    /**
+     * Whether search should be case sensitive.
+     * @type {boolean}
+     */
+    this.caseSensitive = false;
+
+    /**
+     * Whether search should preserve the currently selected block by default.
+     * @type {boolean}
+     */
+    this.preserveSelected = true;
+  }
+
+  /**
+   * Initializes the workspace search bar.
+   */
+  init() {
+    injectSearchCss();
+    this.createDom_();
+    this.setVisible_(false);
+  }
+
+  /**
+   * Creates and injects the search bar's DOM.
+   * @protected
+   */
+  createDom_() {
+    /*
+     * Creates the search bar. The generated search bar looks like:
+     * <div class="ws-search'>
+     *   <div class="ws-search-container'>
+     *     <div class="ws-search-content'>
+     *       <div class="ws-search-input'>
+     *         [... text input goes here ...]
+     *       </div>
+     *       [... actions div goes here ...]
+     *     </div>
+     *     [... close button goes here ...]
+     *   </div>
+     * </div>
+     */
+    const injectionDiv = this.workspace_.getInjectionDiv();
+    injectionDiv.addEventListener('keydown',
+        evt => this.onWorkspaceKeyDown_(/** @type {KeyboardEvent} */ evt));
+
+    this.htmlDiv_ = document.createElement('div');
+    Blockly.utils.dom.addClass(this.htmlDiv_, 'blockly-ws-search');
+    this.positionSearchBar_();
+
+    const searchContainer = document.createElement('div');
+    Blockly.utils.dom.addClass(searchContainer, 'blockly-ws-search-container');
+
+    const searchContent = document.createElement('div');
+    Blockly.utils.dom.addClass(searchContent, 'blockly-ws-search-content');
+    searchContainer.append(searchContent);
```

Comment:
``append`` not available on IE11. 
Babel only transforms syntax. If you have a really good reason to use ``append`` instead of ``appendChild`` then you'll need to add a polyfill.

Reply:
done